File: backend_clickmesa/backend_clickmesa/schemas/shopping_list.py

from datetime import datetime
from typing import List, Optional

from pydantic import BaseModel


class ShoppingListItemBase(BaseModel):
    name: str
    quantity: float
    unit: str
    purchased: bool = False


class ShoppingListItemCreate(ShoppingListItemBase):
    shopping_list_id: int


class ShoppingListItemPublic(ShoppingListItemBase):
    id: int
    shopping_list_id: int

    class Config:
        from_attributes = True


class ShoppingListBase(BaseModel):
    name: str
    owner_id: int

    class Config:
        from_attributes = True


class ShoppingListCreate(ShoppingListBase):
    pass


class ShoppingListUpdate(BaseModel):
    name: Optional[str] = None


class ShoppingListPublic(ShoppingListBase):
    id: int
    created_at: datetime
    items: List[ShoppingListItemPublic]

    class Config:
        from_attributes = True


class ShoppingListList(BaseModel):
    shopping_lists: List[ShoppingListPublic]

class ShoppingListItemCreateInList(BaseModel):
    name: str
    quantity: float
    unit: str
    purchased: bool = False

class ShoppingListCreateWithItems(BaseModel):
    name: Optional[str] = None
    items: List[ShoppingListItemCreateInList]

File: frontend_clickmesa/src/types/shoppingList.ts

export interface ShoppingListItem {
    id:string;
    name: string;
    quantity: number;
    unit: string;
    purchased: boolean;
    shopping_list_id: number;

}

export interface ShoppingList {
    id: string;
    name: string;
    created_at: string;
    owner_id: number;
    items: ShoppingListItem[];  
    
}

export interface NewShoppingListItem {
    name: string;
    quantity: number;
    unit: string;
    purchased: boolean;
}